src/structlog/_base.py

```diff
@@ -9,13 +9,21 @@
 
 from __future__ import annotations
 
+import sys
+
 from typing import Any, Iterable, Mapping, Sequence
 
 from structlog.exceptions import DropEvent
 
 from .typing import BindableLogger, Context, Processor, WrappedLogger
 
 
+if sys.version_info >= (3, 11):
```

Comment:
Any reason why you’re doing this guarded import dance instead of adding Self to the `.typing` import three lines up?

Reply:
I initially tried this in [c04f3ec8](https://github.com/hynek/structlog/pull/659/commits/c04f3ec824e768933fc3c91f86a179557276fa2b#diff-7f889fd0855ea2564d56eaefdf3783b06b62a9d9585281cdebefcd884f027ca2R16), but pre-commit has [a rule](https://github.com/hynek/structlog/actions/runs/11188922225/job/31108545798?pr=659#step:6:29) against importing an attribute that eh module doesn't explicitly export. I tried to get around this a couple times ([fa43344b](https://github.com/hynek/structlog/pull/659/commits/fa43344ba466774afd39464c5a991be96033056f), [9206ef49](https://github.com/hynek/structlog/pull/659/commits/9206ef49bbefed3044aa6d35b387db520b0bef9e)) but I couldn't find a way which worked without violating one pre-commit rule or another. If there's a way to explicitly re-export `typing.Self` from `structlog.typing`, I think it'd be preferable to import it from there.

Reply:
ugh right thanks. Maybe I'll add a _compat module one day, like in attrs. one can re-export using `__all__`